stl/inc/mdspan

```diff
@@ -1131,6 +1131,22 @@ struct _Mdspan_accessor_base<_AccessorPolicy> {
     static constexpr _AccessorPolicy _Acc{};
 };
 
+#if _CONTAINER_DEBUG_LEVEL > 0
+template <class _IndexType, class _OtherIndexType>
+_NODISCARD constexpr _IndexType _Mdspan_checked_index_cast(_OtherIndexType&& _Idx) noexcept(
+    is_nothrow_constructible_v<_IndexType, _OtherIndexType>) {
+    _STL_INTERNAL_STATIC_ASSERT(is_integral_v<_IndexType> && is_constructible_v<_IndexType, _OtherIndexType>);
+
+    using _Arg_value_t = remove_cvref_t<_OtherIndexType>;
+    if constexpr (is_integral_v<_Arg_value_t> && !is_same_v<_Arg_value_t, bool>) {
+        _STL_VERIFY(_STD in_range<_IndexType>(_Idx),
+            "The index must be representable by index_type in order to be in a valid "
+            "multidimensional index (N4964 [mdspan.mdspan.members]/3).");

```

Comment:
This message isn't quite right. It's confusing that we're using "index" and "multidimensional index" with different meanings. How about:
```suggestion
            "Each argument to operator[] must be representable by index_type in order for the pack of arguments to be "
            "a valid multidimensional index (N4964 [mdspan.mdspan.members]/3).");
```